packages/orm/src/relations/ManyToOneReference.ts

```diff
@@ -133,11 +134,16 @@ export class ManyToOneReferenceImpl<T extends Entity, U extends Entity, N extend
     // This should only be callable in the type system if we've already resolved this to an instance,
     // but, just in case we somehow got here in an unloaded state, check to see if we're already in the UoW
     if (!this._isLoaded) {
-      const existing = this.maybeFindEntity();
```

Comment:
Before this failed anytime `existing` was `undefined`, but really if `current` is `undefined`, i.e. the m2o is unset, then its expected to `maybeFindEntity` to also not return anything.